src/Donations/LegacyListeners/DispatchGivePreInsertPayment.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Give\Donations\LegacyListeners;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\Framework\Support\ValueObjects\Money;
+
+/**
+ * Provides support for the legacy give_pre_insert_payment hook
+ *
+ * @unreleased
+ */
+class DispatchGivePreInsertPayment
```

Comment:
@JasonTheAdams does this actually work? I would think this needs to be given to `DispatchGiveInsertPayment` to be functional.

Reply:
Honestly, I haven't tested it. Hahah! I'm hooking into `give_donation_creating` as we want the hook to fire *before* the donation is inserted, not after.

Reply:
I'm still not convinced this will do anything. Would you mind testing this out? Just want to make sure if we're supporting legacy hooks then we're confident they work as expected.